internal/v1/handler_blueprints_test.go

```diff
@@ -95,6 +95,149 @@ func TestHandlers_CreateBlueprint(t *testing.T) {
+	updatedBlueprint, err = BlueprintFromEntry(blueprintEntry)
+	require.NoError(t, err)
+	require.Empty(t, *(*updatedBlueprint.Customizations.Users)[0].Password)
+
+	// "<REDACTED>" value if user didn't have password previously = FAIL
+	body["customizations"] = map[string]interface{}{"users": []map[string]interface{}{{"name": "test", "password": "<REDACTED>", "ssh_key": "ssh key"}}}
+	statusCode, responseBody = tutils.PutResponseBody(t, fmt.Sprintf("http://localhost:8086/api/image-builder/v1/blueprints/%s", result.Id), body)
+	require.Equal(t, http.StatusUnprocessableEntity, statusCode)
+	err = json.Unmarshal([]byte(responseBody), &jsonResp)
+	require.NoError(t, err)
+	require.Equal(t, "Invalid user", jsonResp.Errors[0].Title)
+	require.Equal(t, "Password of user 'test' is redacted and no password found in previous version of blueprint.", jsonResp.Errors[0].Detail)
+
+	// Set password in addition to ssh_key = SUCCESS
+	body["customizations"] = map[string]interface{}{"users": []map[string]interface{}{{"name": "test", "password": "test", "ssh_key": "ssh key"}}}
+	statusCode, _ = tutils.PutResponseBody(t, fmt.Sprintf("http://localhost:8086/api/image-builder/v1/blueprints/%s", result.Id), body)
+	require.Equal(t, http.StatusCreated, statusCode)
+
+	blueprintEntry, err = dbase.GetBlueprint(ctx, result.Id, "000000", nil)
+	require.NoError(t, err)
+
+	updatedBlueprint, err = BlueprintFromEntryWithRedactedPasswords(blueprintEntry)
+	require.NoError(t, err)
+	require.NotNil(t, (*updatedBlueprint.Customizations.Users)[0].Password)
+	require.True(t, (*updatedBlueprint.Customizations.Users)[0].IsRedacted())
+
+	updatedBlueprint, err = BlueprintFromEntry(blueprintEntry)
+	require.NoError(t, err)
+	require.NotNil(t, (*updatedBlueprint.Customizations.Users)[0].Password)
+	require.NotEmpty(t, *(*updatedBlueprint.Customizations.Users)[0].Password)
+	require.NotEqual(t, "test", *(*updatedBlueprint.Customizations.Users)[0].Password) // still hashed
+
+	// "<REDACTED>" value if user had password previously = SUCCESS (pasword unchanged)
+	body["customizations"] = map[string]interface{}{"users": []map[string]interface{}{{"name": "test", "password": "<REDACTED>", "ssh_key": "ssh key"}}}
+	statusCode, _ = tutils.PutResponseBody(t, fmt.Sprintf("http://localhost:8086/api/image-builder/v1/blueprints/%s", result.Id), body)
+	require.Equal(t, http.StatusCreated, statusCode)
+
+	updatedBlueprint, err = BlueprintFromEntryWithRedactedPasswords(blueprintEntry)
+	require.NoError(t, err)
+	require.NotNil(t, (*updatedBlueprint.Customizations.Users)[0].Password)
+	require.True(t, (*updatedBlueprint.Customizations.Users)[0].IsRedacted())
+
+	updatedBlueprint, err = BlueprintFromEntry(blueprintEntry)
+	require.NoError(t, err)
+	require.NotNil(t, (*updatedBlueprint.Customizations.Users)[0].Password)
+	require.NotEmpty(t, *(*updatedBlueprint.Customizations.Users)[0].Password)
+	require.NotEqual(t, "test", *(*updatedBlueprint.Customizations.Users)[0].Password) // still hashed
```

Comment:
improve this check, have to assert the previous password, imho notEqual is not enough